src/core/scene/a-scene.js

```diff
@@ -265,6 +273,17 @@ module.exports.AScene = registerElement('a-scene', {
     },
 
     /**
+     * Handle exiting VR through other means than `<ESC>` key. For example, GearVR back button
+     * on Oculus Browser.
+     */
+    onVRPresentChange: {
+      value: function () {
```

Comment:
this should check `if (evt.display && evt.display.getLayers && evt.display.getLayers().indexOf(self.canvas) === -1) { return; }` so that VR mode isn't toggled for the wrong presentation event (examples: multiple A-Frame scenes on a page, or if there are two VR displays connected but the active A-Frame scene isn't the content being presented)